downloader/artifact_downloader.go

```diff
@@ -183,18 +197,18 @@ func (ad *ConcurrentArtifactDownloader) downloadAndExtractZipArchive(targetDir,
 
 	fileFullPath, details, err := ad.downloadFile(tmpDir, fileName, downloadURL)
 	if err != nil {
-		return "", TransferDetails{}, err
```

Comment:
For some reason I was returning empty `TransferDetails` structs in case of an error. That is wrong because we could return some real data which will indicate how much was downloaded and approximately after how much time the error happened. This could help us spot patterns.